File: frontend/src/data/testData.ts
import { faker } from '@faker-js/faker';
import { generateRandomUser, generateRandomPaymentDetails } from '../utils/helpers';

/**
 * Fixed test user data for scenarios that need consistent data
 */
export const fixedTestUser = {
  name: 'Test User',
  email: 'test.user@example.com',
  password: 'password123',
  title: 'Mr',
  birthDate: '1',
  birthMonth: 'January',
  birthYear: '1990',
  firstName: 'Test',
  lastName: 'User',
  company: 'Test Company',
  address1: '123 Test Street',
  address2: 'Apt 456',
  country: 'United States',
  state: 'California',
  city: 'Los Angeles',
  zipcode: '90001',
  mobileNumber: '1234567890'
};

/**
 * Dynamic test user data that changes on import
 * Use this for tests that need unique user data
 */
export const testUser = generateRandomUser();

/**
 * Payment data
 */
export const paymentDetails = generateRandomPaymentDetails();

/**
 * Product search data
 */
export const searchProducts = {
  validProduct: faker.helpers.arrayElement(['dress', 'top', 'tshirt', 'jeans']),
  invalidProduct: `invalid${faker.string.alphanumeric(10)}`
};

File: backend/data/test_data.py
"""Test data for backend API tests."""
import datetime
from faker import Faker
from utils.helpers import generate_test_user, generate_payment_details

# Initialize Faker
fake = Faker()

class TestData:
    """Test data class."""
    
    # Fixed user data for authentication
    FIXED_USER = {
        "email": "test@example.com",
        "password": "password123",
        "name": "Test User"
    }
    
    # Dynamic user data that will be regenerated when imported
    TEST_USER = generate_test_user()
    
    # Invalid user data
    INVALID_USER = {
        "email": "invalid@example.com",
        "password": "invalidpassword"
    }
    
    # Product search data
    PRODUCT_SEARCH = {
        "valid_product": fake.random_element(elements=["top", "dress", "tshirt", "jeans"]),
        "invalid_product": f"invalid{fake.pystr(min_chars=8, max_chars=12)}"
    }
    
    # Payment details
    PAYMENT_DETAILS = generate_payment_details()
    
    @staticmethod
    def get_dynamic_user():
        """Generate a fresh user for tests requiring unique users.
        
        Returns:
            Newly generated user data
        """
        return generate_test_user()
